src/main/java/de/tum/cit/aet/artemis/programming/service/jenkins/JenkinsXmlFileUtils.java

```diff
@@ -60,7 +60,8 @@ public static Document readXmlFile(Resource resource, @Nullable Map<String, Stri
         }
     }
 
-    private static Document parseDocument(String configXmlText) {
+    // TODO: should we really parse this into a document?
```

Comment:
What would be the alternative?

I‘m not sure if parsing into a DTO is possible here, since the XML configuration of build jobs can contain the additional config of any Jenkins-plugin that has folder/job-specific settings. I.e. we do not know the schema in advance.